### PR TITLE
Diona Species Behavior Handling Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Diona species handling refactor. Zero impacts on balance or existing behavior as far as gameplay/effects go.

Moves Diona's ability behavior out of human life code to their own species life code.

Code wise

- Freed up two species flags.
- Species that are `RADIMMUNE` can now accumulate radiation, though this does absolutely nothing to them (with the exception of Diona).

### CODE REVIEW

**Comment by Aurorablade** [comment]
Why revert the heart thing? I mean ultimatly its all good but just curious.

**Comment by Fox-McCloud** [comment]
>Why revert the heart thing? I mean ultimatly its all good but just curious.

Because ultimately, I wanted this to be as pure of a refactor as possible that changed as little behavior as possible.

**Comment by Aurorablade** [comment]
understood.

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -192,26 +192,15 @@
 
 			if (radiation > 100)
 				radiation = 100
-				if(!(species.flags & RAD_ABSORB))
-					Weaken(10)
-					if(!lying)
-						to_chat(src, "<span class='alert'>You feel weak.</span>")
-						emote("collapse")
+				Weaken(10)
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/human/life.dm]: Doesn't this mean Dionae can get stunned by having 100 rads now?

Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/life.dm]: Nope: https://github.com/ParadiseSS13/Paradise/pull/4188/files#diff-c263363df08edeb8a6e85abb9a411321R625 RADIMMUNE

https://github.com/ParadiseSS13/Paradise/blob/a3fdeaa57cdf5b84dc3be2057c0eb0373a41352a/code/modules/mob/living/carbon/human/life.dm#L181

**Pull request merged**